shell/spy.go

```diff
@@ -1,19 +1,32 @@
 package shell
 
 import (
-	"bytes"
+	"fmt"
+	"io"
 	"log"
 	"os/exec"
+	"strings"
 )
 
-// Execute accepts a command argument and returns the result of the executed command
-func Execute(script string) string {
-	cmd := exec.Command(script)
-	var out bytes.Buffer
-	cmd.Stdout = &out
+// ParseCommand accepts a string and returns the command name and args
+func ParseCommand(cmd string) (string, []string) {
+	stringResult := strings.Split(cmd, " ")
+
+	if len(stringResult) == 1 {
+		return cmd, []string{}
+	}
+	return stringResult[0], stringResult[1:]
+}
+
+// Execute accepts a command argument and writes the result to a writer
+func Execute(writer io.Writer, commandLine string) {
```

Comment:
Since it's so common for functions to take a writer, it's conventionally named `w` (though you're under no obligation to stick to this if you feel it needs a slightly more informative name in context. It's always better to use names that are too long rather than too short, though of course better still is to use names that are _just right_.)